File: LeetCode/02331. Evaluate Boolean Binary Tree/02331.c
bool evaluateTree(struct TreeNode* root){
    if (root->val <= 1) {
        return root->val == 1;
    } else if (root->val == 2) {
        return evaluateTree(root->left) || evaluateTree(root->right);
    } else {
        return evaluateTree(root->left) && evaluateTree(root->right);
    }
}

File: LeetCode/02331. Evaluate Boolean Binary Tree/02331.go
// https://leetcode.com/problems/evaluate-boolean-binary-tree

func evaluateTree(root *TreeNode) bool {
    if root == nil {
        return false
    }

    switch root.Val {
        case 0: // False
            return false

        case 1: // True
            return true

        case 2: // OR
            return evaluateTree(root.Left) || evaluateTree(root.Right)

        case 3: // AND
            return evaluateTree(root.Left) && evaluateTree(root.Right)
    }

    return false
}
